python/cudf/cudf/core/frame.py

```diff
@@ -1819,6 +1827,271 @@ def sample(
+                codes = cudf_indices_frame._data[name]
+                cudf_category_frames[name] = build_categorical_column(
+                    cudf_dictionaries_frame._data[name],
+                    codes,
+                    mask=codes.base_mask,
+                    size=codes.size,
+                    ordered=dict_ordered[name],
+                )
+
+        # Handle non-dict arrays
+        cudf_non_category_frames = (
+            cudf.DataFrame()
+            if data.num_columns == 0
+            else cudf.DataFrame._from_table(
+                libcudf.interop.from_arrow(data, data.column_names)
+            )
+        )
+
+        if (
+            cudf_non_category_frames.shape[1] > 0
+            and cudf_category_frames.shape[1] > 0
+        ):
+            result = cudf.concat(
+                [cudf_non_category_frames, cudf_category_frames], axis=1
+            )
+        elif cudf_non_category_frames.shape[1] > 0:
+            result = cudf_non_category_frames
+        else:
+            result = cudf_category_frames
+
+        if dtypes:
+            for name in result._data.names:
+                dtype = dtypes[name]
+                if dtype == "categorical":
+                    dtype = "category"
+                elif dtype == "date" or dtype == "datetime":
+                    dtype = "datetime64[ms]"
+                elif dtype == "empty":
+                    dtype = "object"
+
+                result._data[name] = result._data[name].astype(dtype)
+
+        return result[column_names]
+
+    @annotate("TO_ARROW", color="orange", domain="cudf_python")
+    def to_arrow(self):
+        """
+        Convert to arrow Table
+
+        Examples
+        --------
+        >>> import cudf
+        >>> df = cudf.DataFrame(
+        ...     {"a":[1, 2, 3], "b":[4, 5, 6]}, index=[1, 2, 3])
+        >>> df.to_arrow()
+        pyarrow.Table
+        a: int64
+        b: int64
+        index: int64
+        """
+
+        data = self.copy(deep=False)
+
+        codes = {}
+        # saving the name as they might get changed from int to str
+        codes_keys = []
+        categories = {}
+        # saving the name as they might get changed from int to str
+        names = self._data.names
+        null_arrays_names = []
+
+        for name in self._data.names:
+            col = self._data[name]
+            if isinstance(col, cudf.core.column.CategoricalColumn,):
+                # arrow doesn't support unsigned codes
+                signed_type = (
+                    min_signed_type(col.codes.max())
+                    if col.codes.size > 0
+                    else np.int8
+                )
+                codes[str(name)] = col.codes.astype(signed_type)
+                categories[str(name)] = col.categories
+                codes_keys.append(name)
+            elif isinstance(
+                col, cudf.core.column.StringColumn
+            ) and col.null_count == len(col):
```

Comment:
Will test with other arrays, earlier this was done only for string column so I just used that case.

Reply:
As per my observation with pandas and pyarrow, this happens only for String type, for rest of them it keeps original dtype and values.
```
>>> sr = pd.Series([None, None, None, None], dtype=pd.Int64Dtype())
>>> sr
0    <NA>
1    <NA>
2    <NA>
3    <NA>
dtype: Int64
>>> pa.Array.from_pandas(sr)
<pyarrow.lib.Int64Array object at 0x7f8842b62910>
[
  null,
  null,
  null,
  null
]

```

Reply:
So in the case of Series <--> Array I guess this makes sense since there isn't additional metadata to direct the type, but in the case of DataFrame <--> Table is there a difference?

Reply:
Even in case DataFrame to Table, it is still Int64
```
>>> pdf = pd.DataFrame({"a":sr})
>>> pdf
      a
0  <NA>
1  <NA>
2  <NA>
3  <NA>
>>> pa.Table.from_pandas(pdf)
pyarrow.Table
a: int64
>>> pa.Table.from_pandas(pdf)['a']
<pyarrow.lib.ChunkedArray object at 0x7f1a81720230>
[
  [
    null,
    null,
    null,
    null
  ]
]
>>> pa.Table.from_pandas(pdf)['a'].chunk(0)
<pyarrow.lib.Int64Array object at 0x7f1a81718c90>
[
  null,
  null,
  null,
  null
]

```

Reply:
Okay then, lets follow suite for now and we can address later if we desire to do so.